src/app/api/authentication.service.ts

```diff
@@ -42,29 +42,99 @@ export class AuthenticationService extends ApiImplementation {
         this.GetOwnUser();
     }
     
-    private GetOwnUser() {
-        this.http.get<ExtendedUser>("/users/me").subscribe((data) => {
-            this.user.next(data);
-            this.tokenStorage.SetStoredUser(data);
+    private GetOwnUser(refreshTokenIfFail: boolean = false) {
+        this.http.get<ExtendedUser>("/users/me").subscribe({
+            error: error => {
+                if (refreshTokenIfFail) {
+                    this.RefreshToken();
+                }
+                else {
+                    const apiError: RefreshApiError | undefined = error.error?.error;
+                    this.bannerService.error("Failed to get your user info", apiError == null ? error.message : apiError.message);
+                }
+            },
+            next: response => {
+                this.user.next(response);
+                this.tokenStorage.SetStoredUser(response);
 
-            console.log(data);
+                console.log(response);
+            },
+        });
+    }
+
+    private RefreshToken() {
+        const oldRefreshToken: string | null = this.tokenStorage.GetStoredRefreshToken();
+
+        // At this point we know the token exists in the local store
+        const request: AuthRefreshRequest = {
+            tokenData: oldRefreshToken!
+        }
+
+        this.http.post<AuthResponse>("/refreshToken", request).subscribe({
+            error: error => {
+                this.ResetStoredInformation();
+
+                const apiError: RefreshApiError | undefined = error.error?.error;
+                this.bannerService.error("Failed to extend session", apiError == null ? error.message : apiError.message);
+            },
```

Comment:
We ignore the API error in legacy code:

```suggestion
            error: error => {
                this.ResetStoredInformation();

                const apiError: RefreshApiError | undefined = error.error?.error;
                console.warn("API error during token refresh:", apiError);
                
                this.bannerService.warn("Session Expired", "Your session has expired, please sign in again.");
            },
```